datatypes/src/primitives/spatial_partition.rs

```diff
@@ -238,6 +238,13 @@ impl SpatialPartition2D {
             ),
         }
     }
+
+    pub fn extend(&mut self, other: &Self) {
+        self.upper_left_coordinate.x = self.upper_left().x.min(other.upper_left().x);
```

Comment:
can you do `self = self.extened(&other)` instead of repeating the logic?

Reply:
[c170029](https://github.com/geo-engine/geoengine/pull/819/commits/c170029eecd40f6e4cc17d086b59812dea8c4f30)